nomad/state/state_store.go

```diff
@@ -5342,6 +5357,28 @@ func (s *StateStore) updateSummaryWithJob(index uint64, job *structs.Job,
 		}
 	}
 
+	for v := uint64(0); v <= job.Version; v++ {
```

Comment:
We'll need to limit the versioned summaries we keep track of; there are use cases where jobs are constantly being updated and will have many thousands of versions, which would be a huge waste of space here.

Reply:
I think the main thing I want out of this is a "latest version", and having them by older versions seemed like a nice feature complete thing but the more I work on it the less interested in old version statuses I am.

Maybe I should just change the gist of this from a VersionedSummary to a LatestVersionSummary?

Reply:
I think I'm going to start by keeping the structure the same and only having a latest version in there. This way if we want to give some control to how many versions we store here later, we can do that without completely changing the format.

Reply:
Ack right, we might have allocs running on a job of the not-latest version that matter. For example in a failed deployment situation. Hm.